### PR TITLE
Fix a couple of issues with color scheme MVVM

### PR DESCRIPTION
**Pull request by PankajBhojwani**
## Summary of the Pull Request
- When 'discard changes' is hit, we re-initialize our list of color scheme view models but forgot to tell xaml about it, this commit fixes that. 
- Make sure to exit rename mode when 'update settings' gets called

## References
color schemes mvvm added in #13179 

## PR Checklist
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Validation Steps Performed
Hitting discard changes doesn't cause an inconsistency with the currently selected scheme anymore

### CODE REVIEW

**Comment by zadjii-msft** [comment]
Sorry for busting spellbot, merging main should fix this

File: src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp [hunk 1]
```diff
@@ -31,6 +31,12 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         _settings = settings;
 
+        // Exit rename mode if we're in it
+        if (_InRenameMode)
+        {
+            InRenameMode(false);
+        }
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp:38]:
```suggestion
            InRenameMode(false);
```
I think the setter already checks if the value was different? I could be wrong though.

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp (hunk 1 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp]: oops you're right! thanks

**Review by zadjii-msft — APPROVED** [review]

**Comment by zadjii-msft** [comment]
@msftbot merge this like yesterday

**Comment by zadjii-msft** [comment]
boo

@msftbot merge this in 1 minute

**Review by DHowett — APPROVED** [review]

**Pull request merged**